tests/Issues/SimplifyEmpty/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Tests\Issues\SimplifyEmpty\Fixture;
+
+class Fixture {
+
+    private ?string $string;
+    private ?string $string2;
+
+    public function getString(): ?string
+    {
+        return $this->string;
+    }
+
+    public function getString2(): ?string
+    {
+        return $this->string2;
+    }
+
+    public function check(): bool
+    {
+        if (empty($this->getString()) || (empty($this->getString2()) && !is_numeric($this->getString2()))) {
+            return false;
+        }
+
+        return true;
+    }
+}
+
+?>
+-----
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Tests\Issues\SimplifyEmpty\Fixture;
+
+class Fixture {
+
+    private ?string $string;
+    private ?string $string2;
+
+    public function getString(): ?string
+    {
+        return $this->string;
+    }
+
+    public function getString2(): ?string
+    {
+        return $this->string2;
+    }
+
+    public function check(): bool
+    {
+        return !($this->getString() === null || $this->getString() === '' || (($this->getString2() === null || $this->getString2() === '') && !is_numeric($this->getString2())));
```

Comment:
the transformation still invalid with skip re-print handling

https://3v4l.org/UHLOP

Reply:
I missread it, the open close parentheses is on the right, seems still invalid https://3v4l.org/ZT7Gm or I just missread it.

I think the rule for `SimplifyDeMorganBinaryRector` actually need to be skipped for this case as it is very hard to read, even correct transform.

Reply:
oh, I see, the transformation is correct, it just hard to read as it direct return after false vs true negated.